syft/frameworks/torch/he/fv/evaluator.py

```diff
@@ -30,14 +33,29 @@ def add(self, op1, op2):
         elif isinstance(op1, PlainText) and isinstance(op2, PlainText):
             return self._add_plain_plain(op1, op2)
 
+        elif isinstance(op1, CipherText) and isinstance(op2, PlainText):
+            return self._add_cipher_plain(op1, op2)
+
         elif isinstance(op1, PlainText) and isinstance(op2, CipherText):
-            return self._add_plain_cipher(op1, op2)
+            return self._add_cipher_plain(op2, op1)
+
+        else:
+            raise TypeError(f"Addition Operation not supported between {type(op1)} and {type(op2)}")
+
+    def sub(self, op1, op2):
+        if isinstance(op1, CipherText) and isinstance(op2, CipherText):
+            return self._sub_cipher_cipher(op1, op2)
 
         elif isinstance(op1, CipherText) and isinstance(op2, PlainText):
-            return self._add_plain_cipher(op2, op1)
+            return self._sub_cipher_plain(op1, op2)
+
+        elif isinstance(op1, PlainText) and isinstance(op2, CipherText):
```

Comment:
Maybe a method for the ifs? to check if both are ciphertexts, different or throw exception.
to reduce the duplicated code.